soc/nordic/nrf54h/power.c

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/sys/poweroff.h>
+#include <zephyr/toolchain.h>
+#include <zephyr/pm/policy.h>
+#include <zephyr/arch/common/pm_s2ram.h>
+#include <hal/nrf_resetinfo.h>
+#include <hal/nrf_lrcconf.h>
+#include <hal/nrf_memconf.h>
+#include <zephyr/cache.h>
+#include <power.h>
+#include "pm_s2ram.h"
+
+static void suspend_common(void)
+{
+
+	/* Flush, disable and power down DCACHE */
+	sys_cache_data_flush_all();
+	sys_cache_data_disable();
+	nrf_memconf_ramblock_control_enable_set(NRF_MEMCONF, RAMBLOCK_POWER_ID,
+					RAMBLOCK_CONTROL_BIT_DCACHE, false);
+
+	if (IS_ENABLED(CONFIG_ICACHE)) {
+		/* Disable and power down ICACHE */
+		sys_cache_instr_disable();
+		nrf_memconf_ramblock_control_enable_set(NRF_MEMCONF, RAMBLOCK_POWER_ID,
+						RAMBLOCK_CONTROL_BIT_ICACHE, false);
+	}
+
+	/* Disable retention */
+	nrf_lrcconf_retain_set(NRF_LRCCONF010, NRF_LRCCONF_POWER_DOMAIN_0, false);
+	nrf_lrcconf_poweron_force_set(NRF_LRCCONF010, NRF_LRCCONF_POWER_DOMAIN_0, false);
+}
+
+void nrf_poweroff(void)
```

Comment:
note: this API is only used by the poweroff module, so there's no need to define it here and expose it in a header. Please, move it to poweroff.c (feel free to have multiple poweroff.c files, eg, one specific to nrf54h maybe)

Reply:
`nrf_poweroff()` calls static function `suspend_common()` which introduces common steps for poweroff and s2ram procedures. I'm not sure how to handle this detail.